crates/router/src/connector/globalpay.rs

```diff
@@ -442,7 +453,14 @@ impl ConnectorIntegration<api::Void, types::PaymentsCancelData, types::PaymentsR
         req: &types::PaymentsCancelRouterData,
         _connectors: &settings::Connectors,
     ) -> CustomResult<RequestContent, errors::ConnectorError> {
-        let connector_req = requests::GlobalpayCancelRequest::try_from(req)?;
+        let amount = connector_utils::convert_amount(
+            self.amount_converter,
+            req.request.minor_amount.unwrap_or_default(),
+            req.request.currency.unwrap_or_default(),
+        )?;
```

Comment:
can we not use unwrap_or_default for amount and currency, instead we can either use map or match convert if the amount and currency both are present, else keep the amount as `None`, and pass it to transformers we can create a new cancel routerdata if required to handle optional amount type

Reply:
minor_amount is in already PaymentsCancelRouterData and both arms will be halndled by unwrap_or_default as it will make amount as None for default case rather matching or maping over them

Reply:
unwrap_or_default will make it `MinorUnit(0)` how can unwarp_or_default() make it None